pmlc/target/x86/pipeline.cc

```diff
@@ -335,6 +336,8 @@ void pipelineBuilderStage2(OpPassManager &pm, const Options &options) {
   pm.addNestedPass<func::FuncOp>(pxa::createMemRefDataFlowOptPass());
   pm.addPass(createCanonicalizerPass());
 
+  pm.addNestedPass<func::FuncOp>(
```

Comment:
why this is not enough? you mention something in the PR summary but it is not clear to me.